stdlib/LinearAlgebra/src/lu.jl

```diff
@@ -139,9 +139,9 @@ function generic_lufact!(A::StridedMatrix{T}, ::Val{Pivot} = Val(true);
         for k = 1:minmn
             # find index max
             kp = k
-            if Pivot
-                amax = abs(zero(T))
-                for i = k:m
+            if Pivot && k < m
```

Comment:
Why not `k <= m`? Otherwise this is not a no-op.

Reply:
Both are correct :) if `k == m` there's only one element left in `A[k:m, k]`, so you don't have to search for the maximum, it's `k`, which is already set in line 141.